lib/rules/quotes.js

```diff
@@ -212,7 +213,7 @@ module.exports = {
 
                 // Directive Prologues.
                 case "ExpressionStatement":
-                    return isPartOfDirectivePrologue(node);
+                    return !astUtils.isParenthesised(sourceCode, node) && isExpressionInOrJustAfterDirectivePrologue(node);
```

Comment:
> However, since the PR is about no-extra-parens, and this change isn't just refactoring, it might be better to extract all changes to the quotes rule into a separate PR?

Agreed. These fixes in the `quotes` rule were not expected from my side, they came more as a side effect of refactoring. I'm going to revert these changes for now and I will submit a new PR to update `quotes` once the changes in `ast-utils.js` are merged.